src/main/java/net/javadiscord/javabot/api/routes/user_profile/UserProfileController.java

```diff
@@ -0,0 +1,110 @@
+@RestController
+public class UserProfileController extends CaffeineCache<Pair<Long, Long>, UserProfileData> implements JDAEntity {
+
+	/**
+	 * The constructor of this class which initializes the {@link Caffeine} cache.
+	 */
+	public UserProfileController() {
+		super(Caffeine.newBuilder()
+				.expireAfterWrite(10, TimeUnit.MINUTES)
+				.build()
+		);
+	}
+
+	/**
+	 * Serves a single users' profile in a specified guild.
+	 *
+	 * @param guildId The guilds' id.
+	 * @param userId  The users' id.
+	 * @return The {@link ResponseEntity}.
+	 */
+	@GetMapping(
+			value = "{guild_id}/{user_id}",
+			produces = MediaType.APPLICATION_JSON_VALUE
+	)
+	public ResponseEntity<String> getUserProfile(
+			@PathVariable(value = "guild_id") String guildId,
+			@PathVariable(value = "user_id") String userId
+	) {
+		Guild guild = getJDA().getGuildById(guildId);
+		if (guild == null) {
+			return new ResponseEntity<>(ApiResponses.INVALID_GUILD_IN_REQUEST, HttpStatus.BAD_REQUEST);
+		}
+		User user = getJDA().retrieveUserById(userId).complete();
+		if (user == null) {
+			return new ResponseEntity<>(ApiResponses.INVALID_USER_IN_REQUEST, HttpStatus.BAD_REQUEST);
+		}
+		try (Connection con = Bot.getDataSource().getConnection()) {
+			// Check Cache
+			UserProfileData data = getCache().getIfPresent(new Pair<>(guild.getIdLong(), user.getIdLong()));
+			if (data == null) {
+				data = new UserProfileData();
+				data.setUserId(user.getIdLong());
+				data.setUserName(user.getName());
+				data.setDiscriminator(user.getDiscriminator());
+				data.setEffectiveAvatarUrl(user.getEffectiveAvatarUrl());
+				// Question of the Week Account
+				QOTWPointsService qotwService = new QOTWPointsService(Bot.getDataSource());
+				QOTWAccount qotwAccount = qotwService.getOrCreateAccount(user.getIdLong());
+				data.setQotwAccount(qotwAccount);
+				// Help Account
+				HelpExperienceService helpService = new HelpExperienceService(Bot.getDataSource());
+				HelpAccount helpAccount = helpService.getOrCreateAccount(user.getIdLong());
+				data.setHelpAccount(HelpAccountData.of(helpAccount, guild));
+				// User Preferences
+				UserPreferenceService preferenceService = new UserPreferenceService(Bot.getDataSource());
+				List<UserPreference> preferences = Arrays.stream(Preference.values()).map(p -> preferenceService.getOrCreate(user.getIdLong(), p)).toList();
+				data.setPreferences(preferences);
+				// User Warns
+				WarnRepository warnRepository = new WarnRepository(con);
+				LocalDateTime cutoff = LocalDateTime.now().minusDays(Bot.getConfig().get(guild).getModerationConfig().getWarnTimeoutDays());
+				data.setWarns(warnRepository.getWarnsByUserId(user.getIdLong(), cutoff));
+				// Insert into cache
+				getCache().put(new Pair<>(guild.getIdLong(), user.getIdLong()), data);
+			}
+			return new ResponseEntity<>(new ApiResponseBuilder().add("profile", data).build(), HttpStatus.OK);
+		} catch (SQLException e) {
+			ExceptionLogger.capture(e, getClass().getSimpleName());
```

Comment:
You could also centralize Spring exception handling using [`ResponseEntityExceptionHandler`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/servlet/mvc/method/annotation/ResponseEntityExceptionHandler.html)